python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1461,7 +1461,7 @@ def create_run(
         check.opt_set_param(asset_selection, "asset_selection", of_type=AssetKey)
         check.opt_set_param(asset_check_selection, "asset_check_selection", of_type=AssetCheckKey)
 
-        if asset_selection is not None or asset_check_selection is not None:
+        if asset_selection is not None or asset_check_selection:
```

Comment:
maybe worth a comment here with historical context on why its important that these two differ?